app/geo/gui/src/ts/twxplore/gui/geo/components/map/Map.tsx

```diff
@@ -97,6 +99,16 @@ const MapImpl: React.FunctionComponent = () => {
     }
     return false;
   };
+
+  const isLoading = (loadingState: {[clickedUri: string]: LoadingState}) => {
```

Comment:
This is a one-liner:
const queryInProgress = Object.keys(state.loadingState).some(clickedUri => state.loadingState[clickedUri].queryInProgress)

Reply:
Use the functional idioms like .map, .find, .some, if you can.